pcdsdevices/slits.py

```diff
@@ -453,6 +455,38 @@ def calc_lightpath_state(
         return super().calc_lightpath_state(xwidth=xwidth_readback,
                                             ywidth=ywidth_readback)
 
+class SlitsBlades(BaseInterface, GroupDevice):
+
+    up = FCpt(IMS,'{self._pv_up}')
+    down = FCpt(IMS,'{self._pv_down}')
+    north = FCpt(IMS,'{self._pv_north}')
+    south = FCpt(IMS,'{self._pv_south}')
+
+    def __init__(self, blade_pvs, **kwargs):
```

Comment:
Another approach is to pass them separately rather than in a single list argument. We have a lot of devices that do this using the `UnrelatedComponent` e.g. https://github.com/pcdshub/pcdsdevices/blob/23a747b0cb4b6d7c3719870e16a57e0cbbdd106a/pcdsdevices/ccm.py#L792-L810 but you can also do this without by making extra arguments.

Make sure you also pass *args into the super().__init__ too or the rest of the device won't work properly.

Reply:
Thanks, that's good to know! The lack of dynamic in Ophyd is really frustrating at times...
After discussing with @tangkong , I however think the syntax overhead is not worth it. I'm going to keep the blades at the top level.